conf/aggregations.go

```diff
@@ -1,79 +1,94 @@
 package conf
 
 import (
+	"errors"
 	"fmt"
 	"regexp"
 	"strconv"
 	"strings"
 
-	"github.com/alyu/configparser"
+	"github.com/grafana/configparser"
 )
 
+var defaultAggregation = Aggregation{
+	Name:              "default",
+	Pattern:           regexp.MustCompile(".*"),
+	XFilesFactor:      0.5,
+	AggregationMethod: []Method{Avg},
+}
+
 // Aggregations holds the aggregation definitions
 type Aggregations struct {
 	Data               []Aggregation
 	DefaultAggregation Aggregation
 }
 
 type Aggregation struct {
-	Name              string
-	Pattern           *regexp.Regexp
-	XFilesFactor      float64
-	AggregationMethod []Method
+	Name              string         // mandatory (I *think* carbon allows empty name, but the docs say you should provide one)
+	Pattern           *regexp.Regexp // mandatory (I *think* carbon allows empty pattern, the docs say you should provide one. but an empty string is still a pattern)
+	XFilesFactor      float64        // optional. defaults to 0.5
+	AggregationMethod []Method       // optional. defaults to ['average']
 }
 
 // NewAggregations create instance of Aggregations
 func NewAggregations() Aggregations {
 	return Aggregations{
-		Data: make([]Aggregation, 0),
-		DefaultAggregation: Aggregation{
-			Name:              "default",
-			Pattern:           regexp.MustCompile(".*"),
-			XFilesFactor:      0.5,
-			AggregationMethod: []Method{Avg},
-		},
+		Data:               make([]Aggregation, 0),
+		DefaultAggregation: defaultAggregation,
 	}
 }
 
 // ReadAggregations returns the defined aggregations from a storage-aggregation.conf file
 // and adds the default
 func ReadAggregations(file string) (Aggregations, error) {
-	config, err := configparser.Read(file)
+	config, err := configparser.ReadFile(file)
 	if err != nil {
 		return Aggregations{}, err
 	}
-	sections, err := config.AllSections()
+	_, sections, err := config.AllSections()
 	if err != nil {
 		return Aggregations{}, err
 	}
 
 	result := NewAggregations()
 
 	for _, s := range sections {
-		item := Aggregation{}
-		item.Name = strings.Trim(strings.SplitN(s.String(), "\n", 2)[0], " []")
-		if item.Name == "" || strings.HasPrefix(item.Name, "#") {
-			continue
+		item := Aggregation{
```

Comment:
perhaps reuse values from `defaultAggregation`?

Reply:
need to be careful to use deep copies so different callers can't affect each other, but yeah good idea. fixed that